credentials/credentials.go

```diff
@@ -109,6 +109,13 @@ type TransportCredentials interface {
 	ServerHandshake(net.Conn) (net.Conn, AuthInfo, error)
 	// Info provides the ProtocolInfo of this TransportCredentials.
 	Info() ProtocolInfo
+	// Clone makes a copy of this TransportCredentials.
+	Clone() TransportCredentials
+	// OverrideServerName overrides the server name of this TransportCredentials,
```

Comment:
OverrideServer overrides the server name used to verify the hostname on the returned certificates from the server. gRPC internals also use it to override the virtual hosting name if it is set. It must be called before dialing. Currently, this is only used by grpclb.